tests/Unit/NumericalAlgorithms/Spectral/Test_SwshCollocation.cpp

```diff
@@ -0,0 +1,130 @@
+  // libsharp types.
+  CHECK(precomputed_collocation  // NOLINT
+            .get_sharp_geom_info()
+            ->pair[0]
+            .r1.stride == expected_stride);  // NOLINT
+  CHECK(precomputed_collocation              // NOLINT
+            .get_sharp_geom_info()
+            ->pair[0]
+            .r2.stride == expected_stride);  // NOLINT
+  CHECK(computed_collocation                 // NOLINT
+            .get_sharp_geom_info()
+            ->pair[0]
+            .r1.stride == expected_stride);  // NOLINT
+  CHECK(computed_collocation                 // NOLINT
+            .get_sharp_geom_info()
+            ->pair[0]
+            .r2.stride == expected_stride);  // NOLINT
+
+  size_t offset_counter = 0;
+
+  sharp_geom_info* computed_sgi = computed_collocation.get_sharp_geom_info();
+  for (auto i : precomputed_collocation) {
+    CHECK(i.offset == offset_counter);
+    CHECK(i.theta == computed_collocation.theta(offset_counter));
+    CHECK(i.phi == computed_collocation.phi(offset_counter));
+    CAPTURE(offset_counter);
+    // check theta values:
+    if (offset_counter < (2 * l_max + 1) * (l_max / 2 + 1)) {
+      CAPTURE(offset_counter / (2 * l_max + 1));
+      CHECK(approx(computed_collocation.theta(offset_counter)) ==
+            computed_sgi  // NOLINT
+                ->pair[offset_counter / (2 * l_max + 1)]
+                .r1.theta);  // NOLINT
+      CHECK(approx(computed_collocation.theta(offset_counter)) ==
+            manual_sgi  // NOLINT
+                ->pair[offset_counter / (2 * l_max + 1)]
+                .r1.theta);  // NOLINT
+    } else {
+      CAPTURE(l_max - (offset_counter / (2 * l_max + 1)));
+      CHECK(approx(computed_collocation.theta(offset_counter)) ==
+            computed_sgi  // NOLINT
+                ->pair[l_max - offset_counter / (2 * l_max + 1)]
+                .r2.theta);  // NOLINT
+      CHECK(approx(computed_collocation.theta(offset_counter)) ==
+            manual_sgi  // NOLINT
+                ->pair[l_max - offset_counter / (2 * l_max + 1)]
+                .r2.theta);  // NOLINT
+    }
+
+    // check phi values:
+    CHECK(approx(computed_collocation.phi(offset_counter)) ==
+          2.0 * M_PI *
+              ((offset_counter % (2 * l_max + 1)) / (2.0 * l_max + 1.0)));
+    CHECK(approx(computed_collocation.phi(offset_counter)) ==
+          2.0 * M_PI *
+              ((offset_counter % (2 * l_max + 1)) / (2.0 * l_max + 1.0)));
+    offset_counter++;
+  }
+}
+
+SPECTRE_TEST_CASE("Unit.NumericalAlgorithms.Spectral.SwshCollocation",
+                  "[Unit][NumericalAlgorithms]") {
+  {
+    INFO("Collocation based on contiguous complex data (Interleaved)");
+    test_spherical_harmonic_collocation<ComplexRepresentation::Interleaved>();
+  }
+  {
+    INFO(
+        "Collocation based on a pair of contiguous blocks representing complex "
+        "data (RealsThenImags)");
+    test_spherical_harmonic_collocation<
+        ComplexRepresentation::RealsThenImags>();
+  }
```

Comment:
please add a test where you call `precomputed_spherical_harmonic_collocation` with `l_max > MAXIMUM_L_MAX` just to see what happens :)